internal/service/apigateway/deployment_test.go

```diff
@@ -612,10 +612,28 @@ resource "aws_lambda_function" "test" {
 
 func testAccStageConfig_deploymentCanarySettings(rName, url string) string {
 	return acctest.ConfigCompose(testAccDeploymentConfig_base(rName, url), `
+resource "aws_api_gateway_stage" "test" {
+  rest_api_id   = aws_api_gateway_rest_api.test.id
+  stage_name    = "prod"
+  deployment_id = aws_api_gateway_deployment.test_main.id
+
+  lifecycle {
+    ignore_changes = [variables, canary_settings]
+  }
+}
+
+resource "aws_api_gateway_deployment" "test_main" {
+  depends_on = [aws_api_gateway_integration.test]
+
+  description = "test-api-gateway"
+  rest_api_id = aws_api_gateway_rest_api.test.id
+}
+
 resource "aws_api_gateway_deployment" "test" {
   depends_on = [aws_api_gateway_integration.test]
 
   rest_api_id = aws_api_gateway_rest_api.test.id
+  stage_name  = aws_api_gateway_stage.test.stage_name
```

Comment:
I don't believe the previous test actually created a canary since `canary_setting` added to input after `CreateDeployment` was already executed. When I try without `stage_name` I get an error `BadRequestException: Invalid deployment content specified.Stage test must already be created before making a canary release deployment`

Reply:
Yeah, it wouldn't have worked. On `aws_api_gateway_deployment`, `stage_description` and `canary_settings` only make sense if you've also specified `stage_name`. If you do specify `stage_name`, it either creates a new stage if it doesn't exist or modifies an existing stage.

So we should have two tests: one where the stage doesn't exist so it creates an unmanaged stage, and one where the stage exists and is modified in-place